conan/tools/cmake/cmakedeps2/targets.py

```diff
@@ -32,6 +32,7 @@ def _context(self):
     @property
     def _template(self):
         return textwrap.dedent("""\
+            include_guard()
```

Comment:
We did the same approach in ``CMakeDeps``.

The code that CMake generate is more complex but in essence it is also an include guard too. I have been investigating trying to understand exactly why of that special include guard approach without success, sounds it could be a legacy thing before CMake 3.10 that introduced ``include_guard``, so I think it is good if we move forward with it as it seems to solve the issue without other potential unexpected side effects.